### PR TITLE
[Portuguese] translate context.md

### PR DESCRIPTION
**Pull request by kr4t0z**

### CODE REVIEW

File: site/docs/pt/guide/context.md [hunk 1]
```diff
@@ -0,0 +1,558 @@
...
+### Atalhos
+
+Existem vários atalhos instalados no objeto de contexto.
+
+| Atalho                | Descrição                                                                                           |
+| --------------------- | ----------------------------------------------------------------------------------------------------|
+| `ctx.msg`             | Obtém o objeto da mensagem, inclusive as mensagens editadas                                         |
+| `ctx.chat`            | Obtém o objeto do chat                                                                              |
+| `ctx.senderChat`      | Obtém o objeto de chat do remetente de `ctx.msg` (para mensagens de canal/grupos anônimos)          |
+| `ctx.from`            | Obtém o autor da mensagem, consulta de callback ou outras coisas                                    |
```
Comment (roziscoding) [inline site/docs/pt/guide/context.md:62]:
```suggestion
| `ctx.from`            | Obtém o autor da mensagem, callback query ou outras coisas                                    |

```

File: site/docs/pt/guide/context.md [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+
+  // Pega entidades de e-mail.
+  const emails = ctx.entities("email");
+
+  // Pega entidades de e-mail e telefone.
+  const phonesAndEmails = ctx.entities(["email", "phone"]);
+});
+```
+
+Ainda, se quiser, pode ignorar `ctx.message`, `ctx.channelPost`, `ctx.editedMessage`, etc e simplesmente usar sempre `ctx.msg` no lugar.
```
Comment (roziscoding) [inline site/docs/pt/guide/context.md:94]:
```suggestion
Ainda, se preferir, você pode ignorar `ctx.message`, `ctx.channelPost`, `ctx.editedMessage`, etc e simplesmente usar sempre `ctx.msg` no lugar.

```

File: site/docs/pt/guide/context.md [hunk 3]
```diff
@@ -0,0 +1,558 @@
...
+
+## Sondagem via Verificações "Has"
+
+O objeto de contexto tem alguns métodos que permitem sondar os dados contidos para coisas específicas.
+Por exemplo, você pode chamar `ctx.hasCommand("start")` para ver se o objeto de contexto contém um comando `/start`.
+É por isso que os métodos são coletivamente chamados _verificações has_.
```
Comment (roziscoding) [inline site/docs/pt/guide/context.md:100]:
```suggestion
É por isso que os métodos são coletivamente chamados de _verificações has_.

```

File: site/docs/pt/guide/context.md [hunk 4]
```diff
@@ -0,0 +1,558 @@
...
+
+::: tip Saber Quando Usar Verificações "Has"
+
+Essa é exatamente a mesma lógica usada por `bot.command("start")`.
+Note que você normalmente deve usar [filter queries](./filter-queries) e métodos similares.
```
Comment (roziscoding) [inline site/docs/pt/guide/context.md:105]:
```suggestion
Note que normalmente é melhor usar [filter queries](./filter-queries) e métodos similares.

```

File: site/docs/pt/guide/context.md [hunk 5]
```diff
@@ -0,0 +1,558 @@
...
+  // Pega o identificador do chat.
+  const chatId = ctx.msg.chat.id;
+  // O texto para responder
+  const text = "Recebi sua mensagem!";
+  // Envia a resposta.
+  await bot.api.sendMessage(chatId, text);
+});
+```
+
+Você pode notar duas coisas que não são ideais sobre isso:
```
Comment (roziscoding) [inline site/docs/pt/guide/context.md:139]:
```suggestion
É possível notar duas coisas que não são ideais aqui:

```

File: site/docs/pt/guide/context.md [hunk 6]
```diff
@@ -0,0 +1,558 @@
...
+  await ctx.reply("Recebi sua mensagem!");
+});
+
+// Ou, ainda mais curto:
+bot.on("message", (ctx) => ctx.reply("Entendi!"));
+```
+
+Da hora! :tada:
+
+Por baixo dos panos, o contexto _já conhece o identificador do chat_ (nomeadamente `ctx.msg.chat.id`), então ele te dá o método `reply` para apenas enviar uma mensagem de volta para o mesmo chat.
```
Comment (roziscoding) [inline site/docs/pt/guide/context.md:165]:
```suggestion
Por baixo dos panos, o contexto _já conhece o identificador do chat_ (ou seja, `ctx.msg.chat.id`), então ele te dá o método `reply` para apenas enviar uma mensagem de volta para o mesmo chat.

```

File: site/docs/pt/guide/context.md [hunk 7]
```diff
@@ -0,0 +1,558 @@
...
+Internamente, `reply` novamente chama `sendMessage` com o identificador do chat pré-preenchido para você.
+
+Consequentemente, todos os métodos no objeto de contexto aceitam objetos de opções do tipo `Other` como explicado [anteriormente](./basics#enviando-mensagens).
+Isso pode ser usado para passar mais configurações para cada chamada de API.
+
+::: tip Funcionalidade de Resposta no Telegram
+Mesmo que o método seja chamado de `ctx.reply` no grammY (e em muitos outros frameworks), ele não utiliza a [funcionalidade de resposta do Telegram](https://telegram.org/blog/replies-mentions-hashtags#replies), onde uma mensagem anterior é vinculada.
+
+Se você verificar o que `sendMessage` pode fazer na [Referência da API de Bots do Telegram](https://core.telegram.org/bots/api#sendmessage), verá várias opções, como `parse_mode`, `disable_web_page_preview` e `reply_to_message_id`.
+Este último pode ser usado para fazer uma mensagem como resposta:
```
Comment (roziscoding) [inline site/docs/pt/guide/context.md:175]:
```suggestion
Este último pode ser usado para enviar uma mensagem como resposta:

```

File: site/docs/pt/guide/context.md [hunk 8]
```diff
@@ -0,0 +1,558 @@
...
+await ctx.reply("^ Esta é uma mensagem!", {
+  reply_to_message_id: ctx.msg.message_id,
+});
+```
+
+O mesmo objeto de opções pode ser passado para `bot.api.sendMessage` e `ctx.api.sendMessage`.
+Use o auto-completar para ver as opções disponíveis diretamente no seu editor de código.
+:::
+
+Naturalmente, todos os outros métodos em `ctx.api` têm atalhos com os valores pré-preenchidos corretos, como `ctx.replyWithPhoto` para responder com uma foto, ou `ctx.exportChatInviteLink` para obter um link de convite para o bate-papo correspondente.
```
Comment (roziscoding) [inline site/docs/pt/guide/context.md:187]:
```suggestion
Naturalmente, todos os outros métodos em `ctx.api` têm atalhos com os valores pré-preenchidos corretos, como `ctx.replyWithPhoto` para responder com uma foto, ou `ctx.exportChatInviteLink` para obter um link de convite para o chat correspondente.

```

File: site/docs/pt/guide/context.md [hunk 9]
```diff
@@ -0,0 +1,558 @@
...
+Se você deseja ter uma visão geral sobre quais atalhos existem, então o auto-completar é seu amigo, juntamente com a [Referência da API do grammY](https://deno.land/x/grammy/mod.ts?s=Context).
+
+Note que você pode não querer reagir sempre no mesmo bate-papo.
```
Comment (roziscoding) [inline site/docs/pt/guide/context.md:190]:
```suggestion
Note que você pode não querer reagir sempre no mesmo chat.

```

File: site/docs/pt/guide/context.md [hunk 10]
```diff
@@ -0,0 +1,558 @@
...
+Nesse caso, você pode simplesmente recorrer ao uso dos métodos `ctx.api` e especificar todas as opções ao chamá-los.
+Por exemplo, se você receber uma mensagem da Alice e quiser reagir enviando uma mensagem para o Bob, então você não pode usar `ctx.reply` porque sempre enviará mensagens para o bate-papo com a Alice.
```
Comment (roziscoding) [inline site/docs/pt/guide/context.md:192]:
```suggestion
Por exemplo, se você receber uma mensagem da Alice e quiser reagir enviando uma mensagem para o Bob, então você não pode usar `ctx.reply` porque sempre enviará mensagens para o chat com a Alice.

```

File: site/docs/pt/guide/context.md [hunk 11]
```diff
@@ -0,0 +1,558 @@
...
+Em vez disso, chame `ctx.api.sendMessage` e especifique o identificador do bate-papo do Bob.
```
Comment (roziscoding) [inline site/docs/pt/guide/context.md:193]:
```suggestion
Em vez disso, chame `ctx.api.sendMessage` e especifique o identificador do chat do Bob.

```

File: site/docs/pt/guide/context.md [hunk 12]
```diff
@@ -0,0 +1,558 @@
...
+::: tip Middlequê?
+Esta seção requer uma compreensão de middlewares, então, caso você ainda não tenha pulado para esta [seção](./middleware), aqui está um resumo muito breve.
+
+Tudo que você realmente precisa saber é que vários handlers podem processar o mesmo objeto de contexto.
+Existem handlers especiais que podem modificar `ctx` antes que qualquer outro handler seja executado, e as modificações do primeiro handler serão visíveis para todos os handlers subsequentes.
+:::
+
+A ideia é instalar middlewares antes de registrar outros listeners.
+Você pode então definir as propriedades que deseja dentro desses middlewares.
+Se você fizer `ctx.nomeDaSuaPropriedadePersonalizada = seuValorPersonalizado` dentro de um handler assim, então a propriedade `ctx.nomeDaSuaPropriedadePersonalizada` estará disponível nos handlers restantes também.
```
Comment (roziscoding) [inline site/docs/pt/guide/context.md:224]:
```suggestion
Se você fizer `ctx.nomeDaSuaPropriedadePersonalizada = seuValorPersonalizado` dentro de um handler, a propriedade `ctx.nomeDaSuaPropriedadePersonalizada` estará disponível nos handlers restantes também.

```

File: site/docs/pt/guide/context.md [hunk 13]
```diff
@@ -0,0 +1,558 @@
...
+
+Para fins de ilustração, vamos supor que você queira definir uma propriedade chamada `ctx.config` no objeto de contexto.
+Neste exemplo, vamos usá-lo para armazenar alguma configuração sobre o projeto para que todos os handlers tenham acesso a ela.
+A configuração tornará mais fácil detectar se o bot está sendo usado por seu desenvolvedor ou por usuários regulares.
+
+Logo após criar seu bot, faça o seguinte:
+
+```ts
+const BOT_DEVELOPER = 123456; // identificador do bate-papo do desenvolvedor do bot
```
Comment (roziscoding) [inline site/docs/pt/guide/context.md:233]:
```suggestion
const BOT_DEVELOPER = 123456; // identificador do chat do desenvolvedor do bot

```

**Review by roziscoding — CHANGES_REQUESTED** [review]

File: site/docs/pt/guide/context.md [hunk 14]
```diff
@@ -0,0 +1,558 @@
...
+
+## Informações Disponíveis
+
+Quando um usuário envia uma mensagem para o seu bot, você pode acessá-la através de `ctx.message`.
+Como exemplo, para obter o texto da mensagem, você pode fazer o seguinte:
+
+```ts
+bot.on("message", async (ctx) => {
+  // `txt` será uma `string` ao processar mensagens de texto.
+  // Será `indefinido` se a mensagem recebida não tiver nenhum texto de mensagem,
```
[the remark this answers is not in the document]
Comment (KnorpelSenf) [reply site/docs/pt/guide/context.md:28]: Smells like AI to me

**Review by roziscoding — APPROVED** [review]

**Comment by roziscoding** [comment]
Why the fuck are lint and fmt failing?

**Comment by CikiMomogi** [comment]
Try update the dependencies and run "npm run docs:fmt"

**Comment by CikiMomogi** [comment]
Oh wait, you already done that. 😅
In that case, update your deno.

**Comment by roziscoding** [comment]
Lint is still failing, but the file has the same structure as the one on main.
What am I missing? 🤔

**Comment by KnorpelSenf** [comment]
@roziscoding from https://github.com/grammyjs/website/actions/runs/6532773242/job/17736607570?pr=906#step:4:33 you can see

```
docs/pt/guide/README.md:1 MD041/first-line-heading/first-line-h1 First line in a file should be a top-level heading [Context: "![grammY](/images/grammY.webp)"]
```

**Comment by KnorpelSenf** [comment]
… so you have to do it like in https://raw.githubusercontent.com/grammyjs/website/main/site/docs/guide/README.md and set

```
<!-- markdownlint-disable first-line-heading -->
```

**Pull request merged**